### ISSUE TITLE
Sub-region is not always changed accordingly when traveling using the automation

### ISSUE THREAD

**Issue by Farigh** [Farigh/pokeclicker-automation#24]
### Describe the bug
Moving to a town in another sub-region in Alola won't update the sub-region accordingly in the mini-map.

### Steps to Reproduce

  - Move to `Alola`
  - Move to the `Akala island` sub-region
  - Select `Hau'oli City` in the 'Go to' list and click the 'Go' button

The sub-region never changed: 
![image](https://user-images.githubusercontent.com/11090416/174865814-04994fec-6d05-4bf7-a8b7-26d0c365ad16.png)

### Expected behavior
The player's sub-region should have been switched to:
![image](https://user-images.githubusercontent.com/11090416/174865971-97200dd5-e36a-4f0a-b66d-41ab69fd778a.png)

**Comment by Farigh** [comment]
Sevii dungeons now have a sub-region, this could be used : https://github.com/pokeclicker/pokeclicker/pull/3136/files#diff-41873e3543c7580adc0b3ee74f3596433e6ca53f0edb1cb39d9af1758140f96eR11-R12

**Comment by Daniferrito** [comment]
This requires changes to the pokecliker code, as the information is not there currently.
I created a PR to add the information:
https://github.com/pokeclicker/pokeclicker/pull/3275

Once that PR is merged, the only required change is two lines in [`Route.js`](https://github.com/Farigh/pokeclicker-automation/blob/master/src/lib/Utils/Route.js#L70):
```js
// Move the player to the region first, if needed
if (town.region != player.region)
{
    MapHelper.moveToTown(GameConstants.DockTowns[town.region]);
    player.region = town.region;
    player._subregion(0);
}
```
to
```js
// Move the player to the region first, if needed
if (town.region != player.region)
{
    MapHelper.moveToTown(GameConstants.DockTowns[town.region]);
    player.region = town.region;
}
player.subregion = town.subRegion;
```

**Comment by Daniferrito** [comment]
Well, they dont want any referente to scripting or similar, so I removed any mention or link on either side. Can't get rid of the embed, though.

If they dont want to accept the PR because it incidentally helps a script, I will simply maintain a list of towns with their correct subregions here. Easier for me, and they will have to fill the subregions on their side some day anyways.

**Comment by Farigh** [comment]
Yes links can never be unset once added.
This is annoying, I happened to link the wrong PR to some issue at y work, it sucks. I don't know why GH never made those removable .....

I hope they won't discard it just because you contribute to this script as well.

**Comment by Daniferrito** [comment]
The PR has been merged into develop. Once it is merged into master, this can be fixed. 

I'll create the PR.
